### DISCUSSION TITLE
Is it possible to perform batch processing with kornia.geometry.homography.find_homography_dlt when the number of keypoints varies between batches?

### DISCUSSION THREAD

**Discussion by Yuki-11** [kornia/kornia#2618] [Questions] [answered]
Is it possible to perform batch processing with [kornia.geometry.homography.find_homography_dlt](https://kornia.readthedocs.io/en/latest/geometry.homography.html) when the number of points varies between batches? The points parameter requires a torch.Tensor of shape [B, N, 2], where N needs to be consistent across batches. In many cases, it's expected that the number of points may vary between image pairs. I'm wondering if there's a way to handle this, such as aligning the number of points to match the maximum N within the batch data, to enable parallel processing of the data in mini-batches.

**Comment by edgarriba** [comment, marked as answer]
a direct way would be setting the `weights` to zero to the points you don't want to include to solve the problem. cc @ducha-aiki

**Reply by Yuki-11** [reply]
Thank you for your answer! I found your method very reasonable.
